sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -1039,6 +1039,126 @@ case class GpuArraysOverlap(left: Expression, right: Expression)
+    val repeatedRhs = explodeRhs(rhs.getBase, lhsWithNull.countElements)
+    withResource(lhsWithNull) { lhsWithNull =>
+      // Construct boolean mask where true values correspond to entries to keep
+      //
+      // lhsWithNull: [[1, 2, null], null, [2, 3]]
+      // repeatedRhs: [1, 1, 1, 2, 2]
+      // boolMask: [[F, T, T], null, [F, T]]
+      val boolMask = constructBooleanMask(lhsWithNull.getChildColumnView(0), repeatedRhs,
+                                          lhsWithNull.getListOffsetsView, lhs.getRowCount)
+      withResource(boolMask) { boolMask =>
+        lhsWithNull.applyBooleanMask(boolMask)
+      }
+    }
+  }
+
+  private def explodeRhs(rhs: ColumnVector, counts: ColumnVector): ColumnVector = {
+    withResource(counts) { counts =>
+      withResource(GpuScalar.from(0, DataTypes.IntegerType)) { zero =>
+        withResource(counts.replaceNulls(zero)) { noNullCounts =>
+          withResource(new Table(rhs)) { table =>
+            table.repeat(noNullCounts).getColumn(0)
+          }
+        }
+      }
+    }
+  }
+
+  private def constructBooleanMask(lhs: ColumnView, rhs: ColumnView, 
+                                   offSets: ColumnView, rowCount: Long): ColumnVector = {
+    withResource(lhs) { lhs =>
+      withResource(rhs) { rhs =>
+        val boolMaskNoNans = lhs.equalToNullAware(rhs)
+        val boolMaskWithNans = if (lhs.getType == DType.FLOAT32 || lhs.getType == DType.FLOAT64) {
+            // Compare NaN values for arrays with float or double type
+            withResource(booleanMaskNansOnly(lhs, rhs)) { boolMaskNansOnly =>
+              withResource(boolMaskNoNans) { boolMaskNoNans =>
+                boolMaskNoNans.or(boolMaskNansOnly)
+              }
+            }
+          } else {
+            boolMaskNoNans
+          }
+        withResource(boolMaskWithNans) { boolMaskWithNans =>
+          withResource(boolMaskWithNans.not) { boolMaskToKeep =>
+            withResource(offSets) { offSets =>
+              boolMaskToKeep.makeListFromOffsets(rowCount, offSets)
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private def booleanMaskNansOnly(lhs: ColumnView, rhs: ColumnView): ColumnVector = {
+    withResource(lhs.isNan) { lhsIsNan =>
+      withResource(rhs.isNan) { rhsIsNan =>
+        lhsIsNan.and(rhsIsNan)
+      }
+    }
+  }
+
+  override def doColumnar(lhs: GpuScalar, rhs: GpuColumnVector): ColumnVector = {
+    withResource(GpuColumnVector.from(lhs, rhs.getRowCount.toInt, lhs.dataType)) { left =>
+      doColumnar(left, rhs)
+    }
+  }
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): ColumnVector = {
+    val lhsBase = lhs.getBase
+    withResource(lhsBase.getListOffsetsView) { offSets =>
+      // Construct boolean mask where true values correspond to elements to remove
+      val boolMaskRemove = withResource(lhsBase.getChildColumnView(0)){ lhsFlatten =>
+        withResource(GpuColumnVector.from(rhs, 1, rhs.dataType)) { rhsColumn =>
```

Comment:
Why not just use `BinaryOp.NULL_EQUALS`? Binary ops will take a Scalar as input.

```
withResource(lhsFlatten.equalToNullAware(rhs.getBase())) { boolMaskRemove =>
  ...
}
```

There is no Null aware not equal or we could skip the not too.

Reply:
Updated to use `equalToNullAware`. Thank you!